src/software/ai/hl/stp/tactic/receiver/receiver_fsm.h

```diff
@@ -0,0 +1,296 @@
+#pragma once
+
+#include "shared/constants.h"
+#include "software/ai/evaluation/calc_best_shot.h"
+#include "software/ai/hl/stp/tactic/dribble/dribble_fsm.h"
+#include "software/ai/hl/stp/tactic/kick/kick_fsm.h"
+#include "software/ai/hl/stp/tactic/move/move_fsm.h"
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/intent.h"
+#include "software/ai/intent/move_intent.h"
+#include "software/ai/passing/pass.h"
+#include "software/geom/algorithms/closest_point.h"
+
+struct ReceiverFSM
+{
+    class OneTouchShotState;
+    class ReceiveAndDribbleState;
+    class UndecidedState;
+
+    struct ControlParams
+    {
+        // The pass to receive
+        std::optional<Pass> pass = std::nullopt;
+    };
+
+    DEFINE_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+    // The minimum proportion of open net we're shooting on vs the entire size of the net
+    // that we require before attempting a shot
+    static constexpr double MIN_SHOT_NET_PERCENT_OPEN    = 0.3;
```

Comment:
wondering how this compares to min_open_angle_for_shot_deg in attacker_tactic_ config.yaml

Reply:
its slightly narrower, and i think the attacker should be wider

in the one-touch scenario, we want to take narrower shots because the enemy will have less time to react (not because of the ball speed but because the ball switches directions half way through)

attacker on the other hand should only shoot if it has a higher confidence that it can score (not insanely high, just a bit more than the receiver)